bin/rector.php

```diff
@@ -47,9 +50,26 @@
     $rectorContainerFactory = new RectorContainerFactory();
     $container = $rectorContainerFactory->createFromBootstrapConfigs($bootstrapConfigs);
 } catch (SetNotFoundException $setNotFoundException) {
-    $invalidSetReporter = new InvalidSetReporter();
-    $invalidSetReporter->report($setNotFoundException);
-    exit(ShellCode::ERROR);
+    $match = Strings::match($setNotFoundException->getMessage(), '#Set (".*") was not found#');
+    if ($match) {
+        $path = realpath(trim($match[1], '"'));
+        if ($path) {
+            $rectorContainerFactory = new RectorContainerFactory();
+            try {
+                $container = $rectorContainerFactory->createFromBootstrapConfigs(
+                    new BootstrapConfigs(new SmartFileInfo($path), [])
+                );
+            } catch (SetNotFoundException $setNotFoundException) {
+                $invalidSetReporter = new InvalidSetReporter();
+                $invalidSetReporter->report($setNotFoundException);
+                exit(ShellCode::ERROR);
+            }
+        }
+    } else {
+        $invalidSetReporter = new InvalidSetReporter();
+        $invalidSetReporter->report($setNotFoundException);
+        exit(ShellCode::ERROR);
+    }
```

Comment:
This should not require a container to boot. Purpose of this is to keep thin bin file.

Better approach might something `MissingRectorRulesReporter`, only focused on sets:
https://github.com/rectorphp/rector/blob/main/src/Reporting/MissingRectorRulesReporter.php

Reply:
I don't understand, as at this point, the custom `Laminas\ServiceManager\Migration\Rector\Set\ValueObject\SetList` is never catched as not registered at 

https://github.com/rectorphp/rector/blob/c815334bed781c555f08dff82ce7861d758e9e7a/packages/Set/RectorSetProvider.php#L30

The get by error message is got a full path of set list constant value.

The `MissingRectorRulesReporter` seems only for reporting, not make the custom `SetList` works.

How to make it work if not re-create the `$container` ? is

Reply:
Tried add both `parameters` and `services`. The `services` part not works, only the parameters part.